java/src/server/persistence/register/Register.java

```diff
@@ -31,16 +30,23 @@ public static IRegister getInstance(){
         return _instance;
     }
 
+
     /**
      * Registers a new Persistence Plugin
      *
-     * @param type Type of Persistence Plugin to register
+     * @param location Location of the plugin to register
+     * @throws PluginExistsException
+     * @throws RegisterPluginException
      */
     @Override
-    public IPersistencePlugin registerPlugin(PersistenceType type){
-        IPersistencePlugin plugin;
-        String path = registry.get(type);
-
-        return null;
+    public IPersistencePlugin registerPlugin(String location) throws PluginExistsException, RegisterPluginException {
```

Comment:
Is the registry not going to "register" plugins? Doesn't the PersistencePlugin object need to be instantiated at some point? (I was thinking the registry was going to take the place of a Class Loader).